imap_processing/codice/codice_l1a.py

```diff
@@ -80,101 +73,59 @@ class CoDICEL1aPipeline:
-        # Generate string of random bits of proper length
-        bit_string = bin(random.getrandbits(length_in_bits))[2:]
-        bit_string = bit_string.zfill(length_in_bits)
-        print(f"\tLength of data in bits: {len(bit_string)}")
-
-        # Convert list of random bits into byte-size list of integers
-        random_bytes_str = [bit_string[i : i + 8] for i in range(0, len(bit_string), 8)]
-        random_bytes_int = [int(item, 2) for item in random_bytes_str]
-        random_bytes = bytes(random_bytes_int)
-        print(f"\tLength of data in bytes: {len(random_bytes)}")
-
-        return random_bytes
-
-    def create_science_dataset(self):
+    def create_science_dataset(self) -> xr.Dataset:
         """Create an ``xarray`` dataset for the unpacked science data.
 
         The dataset can then be written to a CDF file.
 
         Returns
         -------
-        xarray.Dataset
+        xr.Dataset
             xarray dataset containing the science data and supporting metadata
 
         # TODO: Pull out common code and put in codice.utils alongside
         # create_hskp_dataset()
         """
-        # Define the attributes specific to species data
-        species_attrs = dataclasses.replace(
-            cdf_attrs.l1a_science_attrs,
-            depend_0="species",
-            depend_1=None,
-            depend_2=None,
-            catdesc="TBD",
-            fieldname="TBD",
-            label_axis="TBD",
-        ).output()
-
-        epoch_time = xr.DataArray(
-            [
-                launch_time for i in range(len(self.data))
-            ],  # Temporary workaround to get epoch in correct dimensions
+        epoch = xr.DataArray(
+            [calc_start_time(self.packets[0].data["SHCOARSE"].raw_value)],
             name="epoch",
             dims=["epoch"],
             attrs=ConstantCoordinates.EPOCH,
         )
 
-        species_array = xr.DataArray(
-            name="species",
-            data=np.zeros(36864),
-            dims=("species"),
-            attrs=species_attrs,
+        energy = xr.DataArray(
+            np.arange(128),
+            name="energy",
+            dims=["energy"],
+            attrs=cdf_attrs.energy_attrs.output(),
         )
 
         dataset = xr.Dataset(
-            coords={"epoch": epoch_time, "species": species_array},
-            attrs=cdf_attrs.codice_l1a_global_attrs.output(),
+            coords={"epoch": epoch, "energy": energy},
+            attrs=cdf_attrs.l1a_lo_sw_species_attrs.output(),
         )
 
         # Create a data variable for each species
-        for species_data, species_name in zip(self.data, LO_SW_SPECIES_NAMES):
-            data = xr.DataArray(
-                name=species_name,
-                data=list(species_data),
-                dims=("species"),
-                attrs=species_attrs,
+        for species_data, varname, fieldname in zip(
+            self.data, LO_SW_SPECIES_VARNAMES, LO_SW_SPECIES_FIELDNAMES
+        ):
```

Comment:
I wonder if you can set this relation between `self.data` and those two list differently. I don't have a good suggestion at the moment. But something to ponder for later.